### PR TITLE
blockchain-link 1.0.15

### PR DESCRIPTION
**Pull request by szymonlesisz**
#### changes
-   Added `details` to `Transaction` object (vin, vout, size, totalInput, totalOutput - used in suite TransactionDetail)
-   Fixed types in `BlockchainLink` event listeners
-   Move "workersDepenedecies" to regular "dependencies" in package.json
-   Update dependencies

### CODE REVIEW

File: packages/blockchain-link/src/types/events.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-}
+import { BlockEvent, NotificationEvent, FiatRatesEvent } from './responses';
 
 export interface Events {
-    connect: ConnectEvent;
-    disconnect: any;
-    notification: any;
-    block: any;
+    connected: undefined;
+    disconnected: undefined;
```
Comment (keraf) [inline packages/blockchain-link/src/types/events.ts:5]: `undefined` type?

**Review by keraf — CHANGES_REQUESTED** [review]
Looks good! Just a bit confused by the `undefined` types (see comment), not sure why that was done like this.

File: packages/blockchain-link/src/types/events.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/blockchain-link/src/types/events.ts:5]: undefined listener function argument (`listener(event)`), this interface is a key/value mapping (event-type-name: event-listener-param)
https://github.com/trezor/trezor-suite/blob/2aa2d54f0125ca486d04281ad6289cf68d21796e/packages/blockchain-link/src/index.ts#L62

connect and disconnect doesnt send any additional data in emit, therefore `event` argument is undefined
https://github.com/trezor/trezor-suite/blob/2aa2d54f0125ca486d04281ad6289cf68d21796e/packages/blockchain-link/src/index.ts#L253

**Comment by szymonlesisz** [comment]
~converting back to draft, i just realised that i need to add two more fields into `details` (script_sig, witness) - required by new rbf firmware~

fine, blockbook doesnt provide this data (yet) so they will be added in next version

File: packages/blockchain-link/src/types/events.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/blockchain-link/src/types/events.ts:5]: btw this could be useful for typings in `desktopApi`

**Review by keraf — APPROVED** [review]

**Pull request merged**